lib/internal/url.js

```diff
@@ -1377,8 +1377,14 @@ function getPathFromURL(path) {
   return isWindows ? getPathFromURLWin32(path) : getPathFromURLPosix(path);
 }
 
+// We percent-encode % character when converting from file path to URL,
+// as this is the only character that won't be percent encoded by
+// default URL percent encoding when pathname is set.
+const percentRegEx = /%/g;
 function getURLFromFilePath(filepath) {
   const tmp = new URL('file://');
+  if (filepath.includes('%'))
```

Comment:
LGTM.
Just for discussion (since I love __removing__ code), but anyway in such cases I yield to the author.
```js
> process.version
'v8.6.0'
> var filepath = "/path/to/??file%20.js"
> console.time('includes'); for (let i = 0; i < 1e8; ++i) { if (filepath.includes('%')) {global.x = i;} };console.timeEnd('includes')
includes: 12336.674ms
> console.time('regex'); for (let i = 0; i < 1e8; ++i) { global.x = filepath.replace(g, '%25'); };console.timeEnd('regex')
regex: 42668.197ms
> console.time('test'); for (let i = 0; i < 1e8; ++i) { if (g.test(filepath)) {global.x = i;} };console.timeEnd('test')
test: 10211.424ms
>
> var filepath = "/path/to/??file20.js"
> console.time('includes'); for (let i = 0; i < 1e8; ++i) { if (filepath.includes('%')) {global.x = i;} };console.timeEnd('includes')
includes: 13628.660ms
> console.time('regex'); for (let i = 0; i < 1e8; ++i) { global.x = filepath.replace(g, '%25'); };console.timeEnd('regex')
regex: 32146.246ms
> console.time('test'); for (let i = 0; i < 1e8; ++i) { if (g.test(filepath)) {global.x = i;} };console.timeEnd('test')
test: 11212.852ms
```
So (1) `percentRegEx.test` is faster than `filepath.includes('%')` (2) the naked replace is ~2-3 slower for the negative case, which I'm sure is completely drowned out by the actual I/O to read that file.

/CC @bmeurer

Reply:
If we're strictly worried about perf then `indexOf` will blow everything out of the water, last I checked. I think it's a worthwhile change.

Reply:
Ouch. That's a bit sad because that is where we started.
https://github.com/nodejs/node/pull/16128#discussion_r143826064
(I still posit this is not worth the micro optimization)

Reply:
The gap between `String.prototype.includes` and `String.prototype.indexOf` should be 
 alot smaller with V8 6.2: https://chromium-review.googlesource.com/c/v8/v8/+/620150